scripts/tools/cancel_workflows_for_pr.py

```diff
@@ -32,28 +37,93 @@
 
 
 class Canceller:
-    def __init__(self, token):
+    def __init__(self, token: str, dry_run: bool):
         self.api = Github(token)
         self.repo = self.api.get_repo(REPOSITORY)
+        self.dry_run = dry_run
+
+    def check_all_pending_prs(self, max_pr_age_days, required_runs):
+        for pr in self.repo.get_pulls(
+            state="open", sort="created_at", direction="desc"
```

Comment:
I am assuming you already looked into this but can we sort based on last modified.

Essentially if I am pushing an update to a PR I first started working on a week ago it seems like we will no longer cancel the pending things

Reply:
https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-pull-requests seems to say updated exists. I will change the PR.